src/components/Post.jsx

```diff
@@ -0,0 +1,145 @@
+  const [newCommentData, setNewCommentData] = useState(initialCommentData);
+  // data on the person that posted the current comment.
+  const [posterInformation, setPosterInformation] = useState(null);
+
+  useEffect(() => {
+    getAContactByID(postData.contactId, setPosterInformation);
+
+    getAllComments(
+      postData.id,
+      setCommentsList,
+      setLoadingComments,
+      setCommentsError
+    );
+  }, [postData.id, setCommentsList, postData.contactId]);
+
+  console.log(loadingComments);
+  //something is wrong. the loading comments state does never return to false so i had to disable them. but the data is working.
+  //if (loadingComments) return "Loading Comments...";
+  if (commentsError) return "Error while loading comments...";
+
+  const handleInputPostData = (event) => {
+    const postCommentValue = event.target.value;
+
+    setNewCommentData(postCommentValue);
+  };
+
+  const HandleSubmit = (event) => {
+    event.preventDefault();
+    const postRequestData = createCommentPostRequestBody(newCommentData);
+
+    MakeAPICommentPostRequest(postRequestData);
+
+    setNewCommentData(initialCommentData);
+  };
+
+  const createCommentPostRequestBody = (newCommentData) => {
+    const postRequestBody = {
+      postId: postData.id,
+      content: newCommentData,
+      contactId: props.simulatedUserID,
+    };
+    return postRequestBody;
+  };
+
+  //function for API Post Request for the Comments
+  const MakeAPICommentPostRequest = (postRequestData) => {
+    const postRequestOption = {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(postRequestData),
+    };
+    //posts comment to comments API
+    postNewCommentToAPI(
+      postData.id,
+      postRequestOption,
+      setCommentsList,
+      commentsList
+    );
+
+    console.log(postRequestOption);
+  };
+
+  return (
+    <div className="post-box main-post">
+      <div className="post-box-head">
+        <div className="post-head-circle">
+          {posterInformation && (
+            <p className="text">
+              {GetInitalsFromNames(
+                posterInformation.firstName,
+                posterInformation.lastName
+              )}
+            </p>
+          )}
+        </div>
+        {posterInformation && (
+          <h2 className="post-head-user">
+            {posterInformation.firstName + " " + posterInformation.lastName}
+          </h2>
+        )}
+
+        <Link to={`/posts/${postData.id}`} className="post-head-title">
+          {postData.title}
+        </Link>
+      </div>
+      <div className="post-box-body">
+        <p className="post-body-content">{postData.content}</p>
+        <hr className="rounded"></hr>
+      </div>
+      <div className="post-box-comments">
+        <p className="post-comments-header">Comments:</p>
+        {commentsList.map((comment, index) => {
+          return (
+            <CommentElement
+              key={index}
```

Comment:
You're using the array index as a key in the commentsList.map call. While this works, it's generally recommended to use a more stable identifier, like a unique comment ID, to help React identify which items have changed, been added, or been removed